Lib/posixpath.py

```diff
@@ -135,6 +135,35 @@ def splitdrive(p):
     return p[:0], p
```

Comment:
IIUC the intention not to let `splitdrive` wrap `splitroot` for `posixpath` is because this is so simple and therefore faster?
Perhaps:
```suggestion
    # intentionally not wrapping splitroot for performance reasons;
    # use empty slice to match str/bytes type for drive to that of p
    return p[:0], p
```

Reply:
As the docstring mentions, the drive is always empty on POSIX, so the implementation of `splitdrive()` is trivial. I'm not sure if this really needs calling out in a comment personally, nor the fact that we preserve the data type (*all* functions in `os.path` do this!)

Reply:
I'm resolving this thread because I feel the implementation and docstring are sufficiently explanatory. Hope that makes sense!